Signal/src/call/PeerConnectionClient.swift

```diff
@@ -392,16 +433,22 @@ class PeerConnectionClient: NSObject, RTCPeerConnectionDelegate, RTCDataChannelD
         // audioTrack is a strong property because we need access to it to mute/unmute, but I was seeing it
         // become nil when it was only a weak property. So we retain it and manually nil the reference here, because
         // we are likely to crash if we retain any peer connection properties when the peerconnection is released
-        Logger.debug("\(TAG) in \(#function)")
-        audioTrack = nil
-        localVideoTrack = nil
-        remoteVideoTrack = nil
+
+        localVideoTrack?.isEnabled = false
+        remoteVideoTrack?.isEnabled = false
+
+        peerConnection.delegate = nil
+        peerConnection.close()
+        peerConnection = nil
```

Comment:
A lot of the crashes I saw while developing I interpreted as "tried to free (e.g.) audiotrack while tearing down peer connection but unexpectedly found something still had a reference to it." 

I was able to avoid these types of crashes by making sure peerConnection client held no references to these objects *before* rtcPeerConnection was dealloc'd

I would have expected to see fewer crashes if you first nil'd the references to all of the resources "owned" by RTCPeerConnection, e.g. dataChannel, audioSender, AudioTrack, videoSender, etc. **then** nilled peerConnection. If this is working, then fine. But something to keep in mind if we see more crashes here.